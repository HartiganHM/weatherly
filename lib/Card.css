.card {
	flex: 1;
	border: 0.01rem solid ghostwhite;
	margin: 1rem;
}
