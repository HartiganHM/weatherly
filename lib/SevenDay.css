.SevenDay {
	color: #FFF;
  display: flex;
  justify-content: space-between;
  border: 2px solid ghostwhite;
  padding: 20px;
  margin: auto;
}

.day,
.condition,
.highLow,
.spacer {
	display: inline-flex;
}
