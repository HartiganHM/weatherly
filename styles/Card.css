.card {
  flex: 1;
  border: 0.01rem solid ghostwhite;
  margin: 1rem;
  padding: 1rem;
}

.temp {
  margin: 0;
}
