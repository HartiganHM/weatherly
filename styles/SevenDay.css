.SevenDay {
  color: #fff;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
}

.day,
.condition,
.highLow,
.spacer {
  display: inline-flex;
}

.sevenDayHeader {
  color: #fff;
  font-family: "Open Sans";
  font-size: 1.8rem;
  padding: 1rem 0 1rem 1rem;
  text-align: left;
}

@media screen and (max-width: 1010px) {
	.SevenDay {
    justify-content: center;
	}
}

@media screen and (max-width: 815px) {
  .sevenDayHeader {
    text-align: center;
  }
}