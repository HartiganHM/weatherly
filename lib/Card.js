import React from 'react'
import '../styles/Card.css'

const Card = (props) => {
	return (
		<div className='card'>
			<div className='time'>{props.title}</div>
			<img src={props.icon} alt="weather condition" />
			<h2 className='temp'>{props.temp ? props.temp : `${props.high} / ${props.low}`}</h2>
		</div>
	)
}

export default Card;