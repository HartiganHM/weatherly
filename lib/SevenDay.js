import React from 'react';
import Card from './Card.js'
import './SevenDay.css'

const SevenDay = ( sevenDayData ) => {	
	let cleanData = sevenDayData.data.slice(0, 7);
  return (
  	<div className='SevenDay'>
  		{
			cleanData.map( (obj) => {
				return (
					<Card title={obj.day}
						  icon={obj.icon}
						  high={obj.high}
						  low={obj.low}
					/>
				)
			})
		}
  	</div>
  )  
}

export default SevenDay;