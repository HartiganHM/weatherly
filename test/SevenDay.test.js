import React from "react";
import { shallow, mount } from "enzyme";
import SevenDay from "../lib/SevenDay";
import Card from "../lib/Card";

const mockSevenDayData = {
  data: [
    {
      day: "Thursday",
      icon: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      high: "53°",
      low: "37°"
    },
    {
      day: "Friday",
      icon: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      high: "67°",
      low: "49°"
    },
    {
      day: "Saturday",
      icon: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      high: "73°",
      low: "43°"
    },
    {
      day: "Sunday",
      icon: "http://icons.wxug.com/i/c/k/clear.gif",
      high: "59°",
      low: "36°"
    },
    {
      day: "Monday",
      icon: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      high: "58°",
      low: "33°"
    },
    {
      day: "Tuesday",
      icon: "http://icons.wxug.com/i/c/k/snow.gif",
      high: "43°",
      low: "32°"
    },
    {
      day: "Wednesday",
      icon: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      high: "55°",
      low: "36°"
    },
    {
      day: "Thursday",
      icon: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      high: "57°",
      low: "35°"
    },
    {
      day: "Friday",
      icon: "http://icons.wxug.com/i/c/k/clear.gif",
      high: "53°",
      low: "36°"
    },
    {
      day: "Saturday",
      icon: "http://icons.wxug.com/i/c/k/partlycloudy.gif",
      high: "56°",
      low: "34°"
    }
  ]
};

let wrapper = <Card />;

describe("SevenDay", () => {
  it("Should have a className SevenDay", () => {
    const sevenDay = shallow(<div className="SevenDay" />);
    expect(sevenDay.find(".SevenDay").length).toEqual(1);
  });

  it("Should take in sevenDayData with a starting length of 10", () => {
    expect(mockSevenDayData.data.length).toEqual(10);
  });

  it("sevenDayData should get reassigned to cleanData and ends with a length of 7", () => {
    let cleanData = mockSevenDayData.data.slice(0, 7);
    expect(cleanData.length).toEqual(7);
  });

  it("cleanData should map over sevenDayData", () => {
    let cleanData = mockSevenDayData.data.slice(0, 7);
    cleanData.map(obj => {
      return (
        <Card title={obj.day} icon={obj.icon} high={obj.high} low={obj.low} />
      );
    });
    // expect(wrapper.props().data).toBe('something');
  });
});
