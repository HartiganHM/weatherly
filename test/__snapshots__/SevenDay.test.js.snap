// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SevenDay Should have a className SevenDay 1`] = `[Function]`;

exports[`SevenDay cleanData should map over sevenDayData 1`] = `[Function]`;

exports[`SevenDay sevenDayData should create 7 cards 1`] = `[Function]`;
